Updating inference services API response

ymao1 opened the pull request:
❗❗This should not be merged until https://github.com/elastic/kibana/pull/203363 is released to serverless

Resolves https://github.com/elastic/ml-team/issues/1428

## Summary

Updates the `GET _inference/_services` API to follow the new agreed upon response format.

For the new `updatable` field, I followed the guidance from API docs which says `you can modify task_settings, secrets (within service_settings), or num_allocations, depending on the specific endpoint service and task_type you’ve created`, so the only fields with `updatable: true` are secret settings field and `num_allocations`.

ymao1 committed:
Updating inference services API response

ymao1 committed:
Merge branch 'main' of github.com:elastic/elasticsearch into ml-1428

ymao1 committed:
Merging in main

elasticsearchmachine commented:
Hi @ymao1, I've created a changelog YAML for you.

ymao1 committed:
Update docs/changelog/118491.yaml

elasticsearchmachine commented:
Pinging @elastic/ml-core (Team:ML)

ymao1 committed:
Delete docs/changelog/118491.yaml

davidkyle reviewed (approved):
LGTM, I have a couple of suggestions

The output contains lots of entries like `"default_value": null,`. In `SettingsConfiguration#toXContent()` please can you wrap writing default_value in a null check.

The service configurations are not sorted as order they are written depends on the iteration order of a map. Sorting the individual configs by `service` would make the output more readable.

ymao1 committed:
Merge branch 'main' of github.com:elastic/elasticsearch into ml-1428

ymao1 committed:
Skip writing null default value config

ymao1 committed:
Sorting output by service name

ymao1 committed:
Merge branch 'ml-1428' of github.com:ymao1/elasticsearch into ml-1428

ymao1 commented:
> The output contains lots of entries like `"default_value": null,`. In `SettingsConfiguration#toXContent()` please can you wrap writing default_value in a null check.

Updated in https://github.com/elastic/elasticsearch/pull/118491/commits/8e1562d6dcea2853a4eb22bf181292a2045fd98d
 
> The service configurations are not sorted as order they are written depends on the iteration order of a map. Sorting the individual configs by `service` would make the output more readable.

Updated in https://github.com/elastic/elasticsearch/pull/118491/commits/bc1922da6e5f792a5a2b422f34a63396a2abfb85

YulNaumenko reviewed (approved):
LGTM! Tested locally, works as expected.

ymao1 committed:
Merge branch 'main' of github.com:elastic/elasticsearch into ml-1428

ymao1 committed:
Updating Jina AI

Commit:
[CI] Auto commit changes from spotless

ymao1 committed:
Merge branch 'main' of github.com:elastic/elasticsearch into ml-1428

ymao1 committed:
Merge branch 'ml-1428' of github.com:ymao1/elasticsearch into ml-1428
